src/nsls2api/cli/beamline.py

```diff
@@ -1,13 +1,102 @@
 import typer
+from rich.panel import Panel
+from rich.table import Table
 
-app = typer.Typer()
+from nsls2api.cli.utils.api import call_nsls2api_endpoint
+from nsls2api.cli.utils.cli_helpers import auto_help_if_no_command
+from nsls2api.cli.utils.console import console, error
 
+app = typer.Typer(invoke_without_command=True)
 
-@app.command()
-def view(beamline: str):
-    print(f"Viewing Beamline : {beamline}")
+
+@app.callback()
+@auto_help_if_no_command()
+def beamline_callback(ctx: typer.Context):
+    pass  # No need to call anything manually
 
 
 @app.command("list")
 def list_beamlines():
-    print("Listing beamlines...")
+    """
+    Retrieve and display a list of beamlines from the API.
+    """
+    endpoint = "v1/beamlines"
+    response = call_nsls2api_endpoint(endpoint, method="GET")
+
+    # Expecting the API to return a list of beamlines
+    if response is None:
+        error("ERROR: Failed to retrieve beamlines.")
+        raise typer.Exit(code=1)
```

Comment:
Although not a major concern, I think it's usually helpful to use a CONST or ENUM for adding context.

Something like `EXIT_GENERAL_FAILURE` or `ExitCode.GENERAL_FAILURE` perhaps.

Reply:
Yeah, good point - I was just lazy and copied this directly from the Typer docs https://typer.tiangolo.com/tutorial/terminating/#exit-with-an-error